tests/queries/0_stateless/03261_minmax_indices_by_default.sql

```diff
@@ -0,0 +1,93 @@
+SET mutations_sync = 2;
+
+DROP TABLE IF EXISTS tbl1;
+DROP TABLE IF EXISTS tbl2;
+DROP TABLE IF EXISTS tbl3;
+
+CREATE TABLE tbl1
+(
+    key Int,
+    x Int,
+    y Int,
+    INDEX x_idx x TYPE minmax
+)
+ENGINE=MergeTree()
+ORDER BY key
+SETTINGS enable_minmax_index_for_all_numeric_columns = true, enable_minmax_index_for_all_string_columns = true;
+
+INSERT INTO tbl1 VALUES (1,1,1), (2,2,2), (3,3,3);
+
+SELECT 'tbl1 with enable_minmax_index_for_all_numeric_columns and enable_minmax_index_for_all_string_columns enabled';
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+ALTER TABLE tbl1 MODIFY SETTING enable_minmax_index_for_all_numeric_columns = false; -- { serverError READONLY_SETTING }
+ALTER TABLE tbl1 MODIFY SETTING enable_minmax_index_for_all_string_columns = false; -- { serverError READONLY_SETTING }
+ALTER TABLE tbl1 RESET SETTING enable_minmax_index_for_all_numeric_columns; -- { serverError READONLY_SETTING }
+ALTER TABLE tbl1 RESET SETTING enable_minmax_index_for_all_string_columns; -- { serverError READONLY_SETTING }
+
+SELECT 'Add numeric column';
+ALTER TABLE tbl1 ADD COLUMN w Int;
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+SELECT 'After materialize';
+ALTER TABLE tbl1 MATERIALIZE INDEX _idx_minmax_n_w;
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+SELECT 'Drop numeric column';
+ALTER TABLE tbl1 DROP COLUMN w;
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+SELECT 'Add string column';
+ALTER TABLE tbl1 ADD COLUMN s String;
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+SELECT 'Add another string column';
+ALTER TABLE tbl1 ADD COLUMN t String;
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+SELECT 'Drop string column t';
+ALTER TABLE tbl1 DROP COLUMN t;
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+SELECT 'Rename column s to ss';
+ALTER TABLE tbl1 RENAME COLUMN s to ss;
+SELECT name, type, expr, data_compressed_bytes FROM system.data_skipping_indices WHERE table = 'tbl1' AND database = currentDatabase();
+
+CREATE TABLE tbl2
+(
+    key Int,
+    x Int,
+    y Int,
+    INDEX _idx_minmax_n_x x TYPE minmax
```

Comment:
The prefixes for auto-created indices are `_idx_minmax_n` and `_idx_minmax_s`. Two thoughts:

1. I think we can omit `_n` and `_s` within these prefixes and rather use only a single prefix for numeric and string columns. I can't think of a situation where it makes sense to disambiguate string and numeric columns.

2. If we use a single prefix, something like `auto_minmax_index_<column_name>` may be a little bit more descriptive.

3. In l. 61 / 75, the test checks that users cannot add indices of the auto-generated index naming scheme. While such a check is useful, we should check only if the auto-index setting(s) is/are on for the table. If it is off, we don't need to bother further and can allow creating indices with funny names.

Reply:
addressed it